simple-schema.js

```diff
@@ -835,11 +835,14 @@ SimpleSchema.prototype.getDefinition = function(key, propList, functionContext)
 
   // For any options that support specifying a function,
   // evaluate the functions.
-  _.each(['min', 'max', 'minCount', 'maxCount', 'allowedValues', 'optional', 'label'], function (prop) {
+  _.each(['min', 'max', 'minCount', 'maxCount', 'allowedValues', 'label'], function (prop) {
     if (_.isFunction(defs[prop])) {
       defs[prop] = defs[prop].call(functionContext || {});
     }
   });
+  if (functionContext && _.isFunction(defs['optional'])) {
+    defs['optional'] = defs['optional'].call(functionContext || {});
+  }
```

Comment:
It is technically a breaking change either way since you will cause the `optional` function not to be called in cases where it currently is called. I plan to merge this into v2 rather than v1, so breaking changes are fine.

In general the fact that it is called in messageForError is somewhat problematic. If schema definitions can be dynamic based on current validation run, then I think the only solution is to store all of the placeholder values in the error object. (Could store the whole message, but then features like reactive language change would not work.) I will have to give this some thought.